src/umpire/alloc/PosixMemalignAllocator.hpp

```diff
@@ -35,8 +37,11 @@ struct PosixMemalignAllocator {
     UMPIRE_LOG(Debug, "(bytes=" << bytes << ") returning " << ret);
 
     if (ret == nullptr) {
-      UMPIRE_ERROR("posix_memalign( bytes = " << bytes << ", pagesize = " << get_page_size()
-                                              << " ) failed with error = " << err);
+      if (err = ENOMEM) {
+        UMPIRE_ERROR(out_of_memory_error, umpire::fmt::format("posix_memalign( bytes = {}, pagesize = {} ) failed with error = {}", bytes, get_page_size(), err));
+      } else {
+        UMPIRE_ERROR(runtime_error, umpire::fmt::format("posix_memalign( bytes = {}, pagesize = {} ) failed with error = {}", bytes, get_page_size(), err));
```

Comment:
```suggestion
        UMPIRE_ERROR(runtime_error, umpire::fmt::format("posix_memalign( bytes = {}, pagesize = {} ) failed with error = {}", bytes, get_page_size(), strerror(int err)));
```

Reply:
I think think the `strerror(int err)` was a typo, so I resolved this manually.